Add global strictMode config var

stevenvegt opened the pull request:
~~Can be used to put the node in "development" or "production" mode.~~
Adds a strictMode boolean flag and an getter on `core.NutsConfig().InStrictMode()`. This var can be used by modules to check if config variables and combinations are valid. Like: do not accept IRMA demo attributes when strict mode is on.

stevenvegt committed:
Add global environment config var

Can be used to put the node in "development" or "production" mode.
Adds the testify package.

stevenvegt committed:
Add env flag to PrintConfig

woutslakhorst commented:
This will give problems in the future since it will hardcode settings. Isn't it better to just have a production flag which disallows certain values in the config? This will guard you when running production but it'll still give the freedom for every other deployment variant.

naming the environments will probably require pre-production, almost-productions, something-more-than-development, etc. You should still be able to manually configure a node to behave the same as production but without the flag/restrictions.

stevenvegt committed:
Add IsProductionMode method to global config

Hels to determine if the node is running in production mode of not. If so, modules can perform checks on config settings.

stevenvegt committed:
Renamed ProductionMode to StrictMode and made it a bool

Instead of naming deployments like production and developement, just talk about strict/unstrict. This flag itself does nothing but can be used for other modules to test settings. E.g. do not accepts IRMA demo attributes when strict mode is on.
